engine/src/main/java/org/terasology/engine/logic/chat/ChatSystem.java

```diff
@@ -72,12 +72,10 @@ public void onMessage(MessageEvent event, EntityRef entity) {
         ClientComponent client = entity.getComponent(ClientComponent.class);
         if (client.local) {
             Message message = event.getFormattedMessage();
-            if (message.getType() == CoreMessageType.CHAT || message.getType() == CoreMessageType.NOTIFICATION) {
-
+            if (message.getType() == CoreMessageType.CHAT || message.getType() == CoreMessageType.NOTIFICATION
+                    && !nuiManager.isOpen(CHAT_UI) && !nuiManager.isOpen(CONSOLE_UI)) {
                 // show overlay only if chat and console are hidden
-                if (!nuiManager.isOpen(CHAT_UI) && !nuiManager.isOpen(CONSOLE_UI)) {
-                    overlay.setVisible(true);
-                }
```

Comment:
I know that I sometimes use constructs like this to put emphasis on the logical control flow, trusting in the compiler to be smart about the details.

I find long `if` conditions with many parts hard to read and to understand, and one has to put quite a lot thought into breaking it up when changing the control flow. 

Therefore, I'm not a huge fan of some these changes in particular, but I'm not going to argue about this. Let's see how it plays out.

Reply:
I'll leave it as is for now